src/toast/ops/pixels_wcs.py

```diff
@@ -0,0 +1,577 @@
+
+            # Get the flags if needed.  Use the same flags as
+            # detector pointing.
+            flags = None
+            if self.detector_pointing.shared_flags is not None:
+                flags = np.array(ob.shared[self.detector_pointing.shared_flags])
+                flags &= self.detector_pointing.shared_flag_mask
+                n_good = np.sum(flags == 0)
+                n_bad = np.sum(flags != 0)
+
+            center_lonlat = None
+            if self.center_offset is not None:
+                center_lonlat = ob.shared[self.center_offset].data
+
+            # Process all detectors
+            for det in ob.select_local_detectors(detectors):
+                for vslice in view_slices:
+                    # Timestream of detector quaternions
+                    quats = ob.detdata[quats_name][det][vslice]
+                    view_samples = len(quats)
+
+                    # print(f"det {det} quats = {quats}")
+                    theta, phi = qa.to_position(quats)
+                    # print(f"det {det} rad theta, phi = {theta}, {phi}")
+
+                    to_deg = 180.0 / np.pi
+                    theta *= to_deg
+                    phi *= to_deg
+                    # print(f"det {det} deg theta, phi = {theta}, {phi}")
+
+                    world_in = np.column_stack([phi, 90.0 - theta])
+
+                    if center_lonlat is not None:
+                        # print(f"orig = {world_in}")
+                        # print(f"center_lonlat = {center_lonlat}")
+                        world_in[:, 0] -= center_lonlat[vslice, 0]
+                        world_in[:, 1] -= center_lonlat[vslice, 1]
+                        # print(f"final = {world_in}")
+
+                    # print(f"det {det} world_in = {world_in}")
+                    rdpix = self.wcs.wcs_world2pix(world_in, 0)
+                    if flags is not None:
+                        # Set bad pointing to pixel -1
+                        bad_pointing = flags[vslice] != 0
+                        rdpix[bad_pointing] = -1
+                    rdpix = np.array(np.around(rdpix), dtype=np.int64)
+                    # print(f"det {det} rdpix = {rdpix}")
+
+                    ob.detdata[self.pixels][det][vslice] = (
+                        rdpix[:, 0] * self.pix_dec + rdpix[:, 1]
+                    )
+
+                    if self.create_dist is not None:
+                        good = ob.detdata[self.pixels][det][vslice] >= 0
+                        # print(f"det {det} has {np.sum(good)} good pixels")
+                        # print((ob.detdata[self.pixels][det][vslice])[good])
+                        self._local_submaps[
+                            (ob.detdata[self.pixels][det][vslice])[good]
+                            // self._n_pix_submap
+                        ] = 1
+
+    @function_timer
+    def _get_scan_range(self, obs):
+        # FIXME: mostly copied from the atmosphere simulation code- we should
```

Comment:
I agree. Perhaps given an observation, the `Focalplane` could tell it's range?

Reply:
Now tracked in issue #573